### ISSUE TITLE
[regression] Readline history broken since 0a16115

### ISSUE THREAD

**Issue by ibuclaw** [cgdb/cgdb#254]
Since 0a16115, readline_history.txt no longer gets updated.

**Comment by ibuclaw** [comment]
Looks like all calls to `rline_add_history()` were completely removed in the commit.

**Comment by brasko** [comment]
CGDB no longer uses readline to store the commands, in fact, it doesn't even know about the commands. It just sends all the text entered in the gdb window directly to gdb. I think gdb would now use it's own readline and store history that way.

Do you see a path where CGDB should still be storing history somehow?

**Comment by ibuclaw** [comment]
Hmm, yes.  I'm currently stepping though it now, and see that it is just one character at a time.  Text is stored in the `scroller`, but it seems that it would be awkward to pry out each individual line from that.

I guess turning on gdb history would indeed be the way to go then.

https://sourceware.org/gdb/onlinedocs/gdb/Command-History.html

Perhaps the last remnants of cgdb's readline_history.txt support should be removed then, in order to not leave around an empty/unused file.

**Comment by brasko** [comment]
OK. Good point. Yes, we should remove it.

One day, perhaps we can use readline in the status bar, and save that histroy, but that's a task for another day.

**Comment by ibuclaw** [comment]
> I just pushed a fix removing the unused code. With that said, do you consider this case closable?

Yes, I guess so.  Is there documentation for transitioning away from the previous readline_history.txt file?

**Comment by brasko** [comment]
I just had to set this up.  I did something like,
set history filename ~/.cgdb/readline_history.txt
set history save on
I wonder if we should automate this in cgdb?

I just don't want to overwrite the readline settings the user already has setup. Thoughts?

**Comment by brasko** [comment]
@ibuclaw Boy, this is an old one, but since we just integrated your other patch, and we were thinking about readline stuff, do we want to re-evaluate this one?

**Comment by ibuclaw** [comment]
@brasko I fixed the ability to access past history by adding the following to `~/.gdbinit`
```
set history save on
set history filename ~/.gdb_history
```

I think I'd still err on documenting setting up a history file, but I'm not personally bothered if cgdb wrote to its own local history file instead.
